### PR TITLE
Add missing dependencies to compute probabilities function

### PR DESCRIPTION
**Pull request by avolkov-intel**
# Description
Add missing dependency for at_device function and resolve logreg test failures

### CODE REVIEW

**Comment by samir-nasibli** [comment]
Please share stable jobs for this fixes with sklearnex tests deselected

**Comment by avolkov-intel** [comment]
CI: http://intel-ci.intel.com/ef767e7e-f951-f12d-8dba-a4bf010d0e2e

**Comment by samir-nasibli** [comment]
> CI: http://intel-ci.intel.com/ef767e7e-f951-f12d-8dba-a4bf010d0e2e

Here sklearnex branch main branch, where this test is deselected. Please update and share the run with the scikit-learn-intelex spmd logreg test deselected

**Comment by avolkov-intel** [comment]
rerun on rebased branch: http://intel-ci.intel.com/ef7691b7-b396-f153-808d-a4bf010d0e2e

**Comment by avolkov-intel** [comment]
CI with enabled test: http://intel-ci.intel.com/ef769255-f5cf-f10a-99ec-a4bf010d0e2e

File: cpp/oneapi/dal/backend/primitives/objective_function/logloss_dpc.cpp [hunk 1]
```diff
@@ -87,7 +88,8 @@ sycl::event compute_probabilities_sparse(sycl::queue& q,
     const std::int64_t p = parameters.get_dimension(0) - (fit_intercept ? 1 : 0);
 
     auto fill_event = fill<Float>(q, probabilities, Float(1), deps);
-    Float w0 = fit_intercept ? parameters.get_slice(0, 1).at_device(q, 0l) : 0; // Poor perfomance
+    Float w0 =
+        fit_intercept ? parameters.get_slice(0, 1).at_device(q, 0l, deps) : 0; // Poor perfomance
```
Comment (Alexandr-Solovev) [inline cpp/oneapi/dal/backend/primitives/objective_function/logloss_dpc.cpp:92]: if the fill event has no deps with w0 computation looks good

**Review by Alexandr-Solovev — APPROVED** [review]

**Review by samir-nasibli — COMMENTED** [review]
Please rebase and re-run CI checks

**Review by samir-nasibli — CHANGES_REQUESTED** [review]
@avolkov-intel I don't see that sklearnex branch has changes for deselecting required spmd tests for logreg. Please address comments and rebase your branches for the check

**Comment by avolkov-intel** [comment]
CI with enabled test: http://intel-ci.intel.com/ef79c8a7-7928-f1dd-a4c8-a4bf010d0e2d

**Comment by samir-nasibli** [comment]
@avolkov-intel please provide the job with the rebased sklearnex main branch with spmd logreg test deselection + current branch

**Comment by avolkov-intel** [comment]
http://intel-ci.intel.com/ef7a5922-4eeb-f191-958d-a4bf010d0e2d

**Comment by samir-nasibli** [comment]
> http://intel-ci.intel.com/ef7a5922-4eeb-f191-958d-a4bf010d0e2d

@avolkov-intel all spmd sklearnex tests are skipped. Could you please check what it the issue?
```bash
spmd/linear_model/tests/test_logistic_regression_spmd.py sssssssssssssss [ 51%]
ssssssssssssssssssssssssssssssssss    
```

**Comment by samir-nasibli** [comment]
> > http://intel-ci.intel.com/ef7a5922-4eeb-f191-958d-a4bf010d0e2d
> 
> @avolkov-intel all spmd sklearnex tests are skipped. Could you please check what it the issue?
> 
> ```shell
> spmd/linear_model/tests/test_logistic_regression_spmd.py sssssssssssssss [ 51%]
> ssssssssssssssssssssssssssssssssss    
> ```

Wrong place was looking. Looks good now.
Please rebase your branch and merge

**Pull request merged**